tests/artifact_rest/test_api_factory.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Test the factory module."""
+
+
+import httpx
+import pytest
+from fastapi import FastAPI
+from hexkit.protocols.dao import DaoFactoryProtocol
+from hexkit.providers.mongodb.testutils import mongodb_fixture  # noqa: F401
+from hexkit.providers.mongodb.testutils import MongoDbFixture
+
+from metldata.artifacts_rest.api_factory import rest_api_factory
+from metldata.artifacts_rest.artifact_info import ArtifactInfo
+from tests.artifact_rest.test_load_artifacts import load_example_artifact_resources
+from tests.fixtures.artifact_info import EXAMPLE_ARTIFACT_INFOS, MINIMAL_ARTIFACT_INFO
+
+
+@pytest.mark.asyncio
+async def get_example_app_client(
+    dao_factory: DaoFactoryProtocol,
+    artifact_infos: list[ArtifactInfo] = EXAMPLE_ARTIFACT_INFOS,
```

Comment:
If you want to postpone fetching EXAMPLE_ARTIFACT_INFOS, you could use None as default value and fetch it in the function body (see below).

Reply:
I hope with performance improvements to the model_utils, we don't have to worry about that at all. If it is OK with you, could we postpone this optimization?

Reply:
Sure, if it's already fast enough, we can do that later.